### DISCUSSION TITLE
[Feature Request] Allow passing DNS challenge API Key via environment variable

### DISCUSSION THREAD

**Discussion by yurividal** [ZoeyVid/NPMplus#1231] [Q&A] [answered]
This would avoid users having to add the key to every proxy, and also make it more secure.

**Comment by Zoey2936** [comment]
there are a lot dns providers, it would be impossible to have envs for all of them

**Reply by yurividal** [reply]
I agree. But most users will have a single one, and all of them will have some sort of Auth token or key.
So, maybe a single DNS_AUTH_TOKEN variable could be used...

**Reply by Zoey2936** [reply, marked as answer]
some services require two tokens or need other values to be set or have different variations on how the token can be set, etc. It is nearly impossible. Another idea would be reuseable crenditals, but that would be something for the (stuck) php version

**Reply by Zoey2936** [reply]
I've added this to: https://github.com/ZoeyVid/NPMplus/discussions/731
